GVFS/GVFS.Common/GVFSConstants.cs

```diff
@@ -7,6 +7,7 @@ public static partial class GVFSConstants
         public const int ShaStringLength = 40;
         public const int MaxPath = 260;
         public const string AllZeroSha = "0000000000000000000000000000000000000000";
+        public const string ExpandedFolderSha = "X_EXPANDED_FOLDER00000000000000000000000";

```

Comment:
This is no longer a sha. Does that matter? Could we just pick another fake, but actual, sha value?

Reply:
I think I picked the wrong spot to put this constant because this is really specific to the placeholder database (and as you mentioned it's not a sha).  I'm thinking we do something like:

- Leave `AllZeroSha` here (as-is)
- Add `EmptyFolderSha` to `PlaceholderListDatabase`
- Move `ExpandedFolderSha` to `PlaceholderListDatabase` and rename it something like `ExpandedFolderValue`

Optionally `AddAndFlush(string path, string sha)` could be renamed to something like `AddAndFlushFile(string path, string sha)` and we could add `AddAndFlushFolder(string path, bool isExpanded)`.

Reply:
> That all makes sense, but what about the value itself?

The rules for the value:

- Must be 40 characters long (due to validation in `PlaceholderListDatabase.TryParseAddLine`)
- Must not be an actual valid SHA (otherwise we risk a file's SHA matching `ExpandedFolderValue` and VFSForGit incorrectly thinking the entry is for a folder)

We have a lot of freedom within those rules, is there a different string you'd prefer?

Reply:
Isn't `0000000000000000000000000000000000000000` a valid sha? It seems like we'd have just as much chance of hitting `0000000000000000000000000000000000000001` as all-zeros, yea?

Reply:
I believe that when @kewillford first added folders to the placeholder list he confirmed that the all-zero SHA will never be a valid SHA for a git blob.  @kewillford, can you confirm?

(And if it *could*, then I'd argue we're better off adding an upgrade step to convert those entries to a different invalid SHA rather than using 00...0001 for expanded folders)

Reply:
Yep, why have a value here that even looks like a sha, if we want it to _not_ be a sha?

Reply:
> There is a check in the git code when writing out the index that will error out if there is a null sha found in the index unless an environment variable is set. That said, not using something that looks like a sha at all for these seems like a better fit for our case especially now that there is more that one.

@kewillford @sanoursa, any objections to adding a major upgrade step to convert the all-zero SHAs to something that doesn't look like a sha?

Reply:
I think that's a good idea. And I would request that these special values that don't look like they are part of the path when manually looking at the PlaceholderList file. (i.e. put some delimiter char(s) at the beginning of the value)

Reply:
> I think that's a good idea. And I would request that these special values that don't look like they are part of the path when manually looking at the PlaceholderList file. (i.e. put some delimiter char(s) at the beginning of the value)

Sounds good, I'll start the values with `@`